File: FlexServ Frontend/FlexServ_Frontend/src/api/userService.js
import API from './adminApi';

// --- User Profile APIs ---
export const getUserProfile = async () => {
  const response = await API.get('/users/profile');
  return response.data;
};

export const updateUserProfile = async (userData) => {
  const response = await API.put('/users/profile', userData);
  return response.data;
};

// --- Address APIs ---
export const getUserAddresses = async () => {
  const response = await API.get('/users/addresses');
  return response.data;
};

export const addAddress = async (addressData) => {
  const response = await API.post('/users/addresses', addressData);
  return response.data;
};

export const updateAddress = async (addressId, addressData) => {
  const response = await API.put(`/users/addresses/${addressId}`, addressData);
  return response.data;
};

export const deleteAddress = async (addressId) => {
  const response = await API.delete(`/users/addresses/${addressId}`);
  return response.data;
};

File: FlexServ Backend/SpringBoot/BusinessService/src/main/java/com/flexserv/service/UserService.java
package com.flexserv.service;

import java.util.List;
import com.flexserv.dto.request.AddressRequest;
import com.flexserv.dto.request.UpdateUserRequest;
import com.flexserv.dto.response.AddressResponse;
import com.flexserv.dto.response.UserResponse;

public interface UserService {

    // User Profile CRUD
    UserResponse getUserProfile(Long userId);
    UserResponse updateUserProfile(Long userId, UpdateUserRequest request);
    void deleteUserAccount(Long userId);

    // Address CRUD
    AddressResponse addAddress(Long userId, AddressRequest request);
    List<AddressResponse> getUserAddresses(Long userId);
    AddressResponse updateAddress(Long userId, Long addressId, AddressRequest request);
    void deleteAddress(Long userId, Long addressId);
}